### PR TITLE
Implement backend to partially address #3

### LINKED ISSUE #3: Cannot resign if it's not your turn
Self-explanatory. This will likely involve changing the `chess.hoon` logic, as (if I recall correctly) the current implementation treats resigning as a move, and you cannot perform moves on your opponent's turn.

### PR DESCRIPTION
**Pull request by bonbud-macryg**
See description at the [this commit](https://github.com/ashelkovnykov/urbit-chess/commit/2454822ad5622e51045f9917ad078050fe422da8) for what I'm sure/not sure about in the overall implementation, and desc. on the [`%end` move commit](https://github.com/ashelkovnykov/urbit-chess/commit/d6a4388bbfa02fa49f39799e065b264b6092a558) for a quick note on that.

### CODE REVIEW

**Comment by ashelkovnykov** [comment]
Will refrain from reviewing until after #31 is merged

File: src/urbit/app/chess.hoon [hunk 1]
```diff
@@ -257,6 +257,43 @@
             ::  record that draw offer is gone
             games  (~(put by games) game-id.action u.game(got-draw-offer |))
           ==
+        %resign
+          =/  game-state  (~(get by games) game-id.action)
+          ::
+          ::  we received a resignation with a game id,
+          ::  do we have a game with that id?
+          ?~  game-state
+            ::  if not, error!
```
Comment (bonbud-macryg) [inline src/urbit/app/chess.hoon:266]: As mentioned in the chat, I have a note to address the "old-style" comments in this PR in line with your previous reviews.

**Comment by ashelkovnykov** [comment]
In a vacuum, this is a good backend commit for issue #3. However, the changes by @rovmug-ticfyn in #40 might also solve this issue in a way that would make the current design in this commit (`%resign`) obsolete. For now, let's leave this PR open and return to it once #40 is merged.

**Comment by ashelkovnykov** [comment]
@bonbud-macryg The work in #40 is basically a superset of this PR, plus work for #4 and #5. Would you prefer:
- We close this PR and open a new one to implement #3 after #40 is done
- We keep this PR open and rebuild on top of #40
- We close this PR and @rovmug-ticfyn can add code for #3 into #40, you keep going with `%telos`

Your call, all are doable!

**Comment by bonbud-macryg** [comment]
If @rovmug-ticfyn is happy to I'd say we close this PR and he can implement a version of this code in #40. When I was initially divvying up the issues I was partly basing it on which could be bundled together and felt like discrete projects, so by now he knows that part of the codebase very well.

**Comment by rovmug-ticfyn** [comment]
Fine with me!

**Pull request closed without merging**